server/src/main/java/org/elasticsearch/common/Rounding.java

```diff
@@ -515,6 +516,7 @@ public Prepared prepare(long minUtcMillis, long maxUtcMillis) {
         }
 
         private TimeUnitPreparedRounding prepareOffsetOrJavaTimeRounding(long minUtcMillis, long maxUtcMillis) {
+//            long minLookup = minUtcMillis - 2*unit.extraLocalOffsetLookup();
```

Comment:
in fact, this could be enough

Reply:
I think because this approach is more clear we should go with it.
The perfomance should not be affected because we only increase this by one unit and we use binary search anyway

Reply:
I think this'd be fine because I think it *might* find an extra DST boundary and that's it. Often it won't find any extra boundaries. We don't even use the time transition boundaries in the fastest case - that case preculcuates all the transitions using the unmodified bounds.

I think it's worth a long comment, but I think it's fine.